### PR TITLE
(ament_xmllint) add extensions argument

### LINKED ISSUE #455: (ament_xmllint) add extension argument
Allow for the extensions to be set. So xacro files etc can also be checked.

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Fixes #455

### CODE REVIEW

**Comment by MatthijsBurgh** [comment]
friendly ping @mjeronimo

Or maybe @clalancette @rodericktaylor?

File: ament_xmllint/ament_xmllint/main.py [hunk 1]
```diff
@@ -39,15 +39,21 @@ def main(argv=sys.argv[1:]):
         nargs='*',
         default=[os.curdir],
         help='The files or directories to check. For directories files ending '
```
Comment (clalancette) [inline ament_xmllint/ament_xmllint/main.py:41]:
```suggestion
        help='The files or directories to check. For directories, only files ending '
```

(I know you didn't add this, but I think we can improve the wording a bit while we are here)

File: ament_xmllint/ament_xmllint/main.py [hunk 2]
```diff
@@ -39,15 +39,21 @@ def main(argv=sys.argv[1:]):
...
-             'in %s will be considered.' %
-             ', '.join(["'.%s'" % e for e in extensions]))
+             'in %s will be considered. Unless overruled by the --extensions '
+             'argument' %
```
Comment (clalancette) [inline ament_xmllint/ament_xmllint/main.py:43]:
```suggestion
             'in %s will be considered (unless overruled by the --extensions '
             'option)' %
```

File: ament_xmllint/ament_xmllint/main.py [hunk 3]
```diff
@@ -39,15 +39,21 @@ def main(argv=sys.argv[1:]):
...
         '--exclude',
         nargs='*',
         default=[],
         help='Exclude specific file names and directory names from the check')
     # not using a file handle directly
     # in order to prevent leaving an empty file when something fails early
+    parser.add_argument(
+        '--extensions',
+        nargs='*',
+        default=default_extensions,
```
Comment (clalancette) [inline ament_xmllint/ament_xmllint/main.py:55]:
```suggestion
        default=['xml'],
```

(and then we can remove the line of code at the beginning of this function)

**Review by clalancette — CHANGES_REQUESTED** [review]
Overall, I think this is a fine change to have.  I've left some feedback for improvement.

File: ament_xmllint/ament_xmllint/main.py (hunk 3 above)
Comment (MatthijsBurgh) [reply ament_xmllint/ament_xmllint/main.py:55]: No, as `default_extensions` is also used to generate the help of the `paths` option.

I don't think we want to maintain that list in two spots. Or we remove the extensions from that help msg.

**Review by clalancette — APPROVED** [review]
Looks reasonable to me, thanks.  I'll run CI on it next.

**Comment by clalancette** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19676/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14183/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20400/)

**Pull request merged**